plugins/ksp/testData/api/referenceElement.kt

```diff
@@ -0,0 +1,38 @@
+// WITH_RUNTIME
+// TEST PROCESSOR: ReferenceElementProcessor
+// EXPECTED:
+// KSClassifierReferenceImpl: Qualifier of B is A
+// KSClassifierReferenceImpl: Qualifier of C is A
+// KSClassifierReferenceImpl: Qualifier of Int is null
+// KSClassifierReferenceImpl: Qualifier of String is null
+// KSClassifierReferenceDescriptorImpl: Qualifier of B is A
+// KSClassifierReferenceDescriptorImpl: Qualifier of C<Int> is A<String>
+// KSClassifierReferenceDescriptorImpl: Qualifier of Int is null
+// KSClassifierReferenceDescriptorImpl: Qualifier of String is null
+// KSClassifierReferenceJavaImpl: Qualifier of H is J
+// KSClassifierReferenceJavaImpl: Qualifier of I is J
+// KSClassifierReferenceJavaImpl: Qualifier of Object is null
+// END
+
+// FILE: reference.kt
+class A<T1> {
+    class B
+    inner class C<T2>
+}
+
+val x: A.B = A.B()
+val y: A<String>.C<Int> = A<String>().C<Int>()
+
+// FILE: J.java
```

Comment:
Java class has to match their file name and obey one class per file, there might be some silent failing if you just put two Java classes in a single file since we are not compiling Java files in our test.

Reply:
It's only required when the Java class is public.